src/wallet.rs

```diff
@@ -301,6 +303,47 @@ impl<F: SecureWalletFile + Debug> Wallet<F> {
         self.wallet.is_some()
     }
 
+    /// Fetches the notes from the state.
+    pub fn get_all_notes(
```

Comment:
The problem with structuring it like this is that *all notes will have to be in memory*. This entails quite the cost to target machine over time. Is it possible to do it differently?

Reply:
Then again, we also do this for making a transaction in `wallet-core`, so this should probably be solved there.

Reply:
You're right

I believe that a further optimization (like iterator implementation over a cache) would be a good approach to follow in a separate PR

We can even batch the process grouping by blockheight (so to have a sort of live data display)